subprojects/core/src/main/java/org/gradle/internal/service/scopes/GlobalScopeServices.java

```diff
@@ -397,4 +402,16 @@ GradleUserHomeScopeServiceRegistry createGradleUserHomeScopeServiceRegistry(Serv
     TimeProvider createTimeProvider() {
         return new TrueTimeProvider();
     }
+
+    MemoryInfo createMemoryInfo(ExecHandleFactory execHandleFactory) {
+        return new MemoryInfo(execHandleFactory);
+    }
+
+    MemoryStatusBroadcaster createMemoryStatusBroadcaster(MemoryInfo memoryInfo, ScheduledExecutorService scheduledExecutorService, ListenerManager listenerManager) {
```

Comment:
The `ScheduledExecutorService` service is provided by `launcher`. Should move its definition to this class, to make it a global service and also add some code to shut it down correctly when the global services are stopped. Or, perhaps better, would be to add something to `ExecutorFactory`.